### ISSUE TITLE
Shift-enter overwrites first workspace instead of saving current

### ISSUE THREAD

**Issue by pjeby** [nothingislost/obsidian-workspaces-plus#47]
**Describe the bug**
Shift-enter Overwrites first workspace instead of saving current

**To Reproduce**
1. Go to a workspace that isn't the first one
2. Click on the indicator -- the current workspace is shown selected in the picker
3. Press shift-enter

**Expected behavior**
The *selected* workspace should be saved

Actual behavior: the first workspace alphabetically is overwritten!

**Desktop (please complete the following information):**
 - OS: Windows

**Additional context**
The context is that I wanted to save my current workspace (which was modified).  It would be nice to have an autosave option and/or an explicit save command to just save the current workspace, instead of allowing it to become out of date.  The way things currently work, it's like they're *snapshots* of how the workspace existed at a point in time, rather than actual persistent workspaces.

**Comment by jsmorabito** [comment]
@pjeby  I'm also seeing some unusual behavior, will try to replicate your situation and document some of what im seeing as well this week. There is a setting in the plugin options that auto saves on switch. is that the behavior you are looking for? 
<img width="400" alt="image" src="https://user-images.githubusercontent.com/46250921/133912338-fcf19035-2aed-48e7-a407-b801935d9566.png">

**Comment by pjeby** [comment]
No, I mean autosaving when the workspace is *modified*, so you don't have to switch to save.  Just like the default workspace autosaves after layout changes.

**Comment by nothingislost** [comment]
@pjeby the modal selection behavior exists because of a request to have the modal behave the same way as the default Obsidian modal, which selects the first item in the list, by default. I'm open to changing the default behavior back if we get agreement that selecting the active workspace is more intuitive.

The shortcut to do what you're after is currently to shift-click directly on the status bar icon/text. That will save the current workspace with no need to open the modal.

I agree that an auto-save on layout change would be a better replacement for the auto-save on switch option. @jsmorabito  do you agree that we can replace the current option with auto-save on layout change?

**Comment by jsmorabito** [comment]
"I agree that an auto-save on layout change would be a better replacement for the auto-save on switch option. @jsmorabito do you agree that we can replace the current option with auto-save on layout change?"

Yup! I also agree with this. Thanks for pointing this out @pjeby. i overlooked this native workspaces behavior and it is consistent with comparable models (ie safari tab groups, mac desktops), they all auto save changes. 

I do think this should be the default behavior; however, we should keep an option to turn it off for use cases such as Murf's 

> My only problem is sometimes I mess around in a workspace and then when I go to load a different one I do not want to save / overwrite the current one. Not sure how to handle but basically would like an ability to load a new workspace without saving current one. [link to discord message](https://discord.com/channels/686053708261228577/840286264964022302/884968025286139935)

I think this ideally would be a per workspace setting and not a global one, but that's a more complicated issue I imagine

So to summarize:
- [ ] change default behavior to autosave workspace changes
- [ ] "Always save current workspace on workspace switch" option can be removed as it is unnecessary now
- [ ] add option to turn off autosave workspaces

**Comment by nothingislost** [comment]
Obsidian never auto saves to a "named" workspace as far as I'm aware. It auto saves the ephemeral "current" workspace which is technically not named. Default Obsidian behavior is to create a snapshot of the "current" workspace state whenever you explicitly request a save.

As a result, I'm not comfortable making auto-save to a named workspace the default functionality but we can certainly make it an option, which is off by default.

**Comment by jsmorabito** [comment]
So the behavior we want is auto save current workspace then? What would it mean to save to a named workspace? I'm not sure I understand the functional difference.

**Comment by nothingislost** [comment]
What Obsidian does, by default, is auto save the current layout to an ephemeral 'current' state which gets automatically restored when you close and reopen the application. This state does not get auto saved into any specific named user workspace automatically. You can see this by loading a workspace, changing it, and then reopening that same workspace. Your changes will not be reflected because you just reloaded the old snapshot without saving first.

The proposed feature is "Auto save layout changes to the active workspace name". Obsidian will not do this by default, and it's a divergence from default behavior, so I'm just saying this won't be the default behavior. It'll be an option the user has to turn on.

**Comment by jsmorabito** [comment]
> What Obsidian does, by default, is auto save the current layout to an ephemeral 'current' state which gets automatically restored when you close and reopen the application. This state does not get auto saved into any specific named user workspace automatically. You can see this by loading a workspace, changing it, and then reopening that same workspace. Your changes will not be reflected because you just reloaded the old snapshot without saving first.
> 
> The proposed feature is "Auto save layout changes to the active workspace name". Obsidian will not do this by default, and it's a divergence from default behavior, so I'm just saying this won't be the default behavior. It'll be an option the user has to turn on.

Just tried that out and it makes sense now. Thanks for clarifying!

**Comment by nothingislost** [comment]
When you get a chance, let us know if the 0.2.0 release sorts out your concerns @pjeby
